### ISSUE TITLE
[postgres] Support more GRANT privileges

### ISSUE THREAD

**Issue by ancoron** [sqlfluff/sqlfluff#350]
Please support at least the following additional `GRANT` privileges for the PostgreSQL dialect:
* `EXECUTE` (e.g. `ON FUNCTION ...`)
* `USAGE` (e.g. `ON SCHEMA ...`)
* `<role_name> TO <role_specification>`

For more details: https://www.postgresql.org/docs/current/sql-grant.html

Simple examples:
```sql
GRANT USAGE ON SCHEMA api TO web_anon;
GRANT EXECUTE ON FUNCTION api.test TO web_anon;
GRANT web_anon TO my_user;
```

...results into:
```
== [db/test.sql] FAIL
L:   1 | P:   1 |  PRS | Found unparsable section: 'GRANT USAGE ON SCHEMA api TO
                       | web_anon'
L:   2 | P:   1 |  PRS | Found unparsable section: 'GRANT EXECUTE ON FUNCTION
                       | api.test TO we...'
L:   3 | P:   1 |  PRS | Found unparsable section: 'GRANT web_anon TO my_user'
```

Examples actually derived from [PostgREST docs](http://postgrest.org/en/v7.0.0/tutorials/tut0.html#step-4-create-database-for-api).

**Comment by alanmcruickshank** [comment]
Agreed. Thanks for the examples too.

The additional base permissions should be very straightforward for this one. We'll need to add some additional segments to handle the roles and role specification but should be fairly straightforward.

**Comment by creste** [comment]
@alanmcruickshank  - I see #1032 was merged and released but this issue hasn't been closed.  Are `GRANT` privileges supposed to work correctly now?  If not, then you can probably ignore the rest of this comment.

If they _are_ supposed to work correctly, then sqlfluff 0.6.4 is still failing to parse some `GRANT`s correctly.

Given this SQL file:
```sql
GRANT CONNECT ON DATABASE mydb TO app;
```
I see this output:
```
> sqlfluff lint --verbose .\test_grant.sql
==== sqlfluff ====
sqlfluff:                0.6.4 python:                  3.9.6
implementation:        cpython dialect:              postgres
verbosity:                   1 templater:               jinja

==== readout ====

=== [ path: .\test_grant.sql ] ===

== [test_grant.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'GRANT
                       | CONNECT ON DATABASE mydb TO app'
==== summary ====
violations:        1 status:         FAIL
All Finished 📜 🎉!
```

If I change this:
```sql
GRANT CONNECT ON DATABASE mydb TO app;
```
to this:

```sql
GRANT ALL ON DATABASE mydb TO app;
```
it works. So I don't think `CONNECT` is supported.

**Comment by barrywhart** [comment]
My guess is that #1032 only addressed part of this issue. This issue is very broad, and will probably be easier for contributors/maintainers to manage if it were split up into smaller issues.

At the moment, the project has a lack of contributors focused on Postgres functionality. Are you potentially interested in and available to do some of this work, @creste?

**Comment by creste** [comment]
@barrywhart  - I don't think I have time to take on any long-term commitments regarding Postgres, but I did have time to try fixing this issue.  Please see https://github.com/sqlfluff/sqlfluff/pull/1339.

**Comment by barrywhart** [comment]
Thanks, @creste!

I know how you feel re: commitments -- I have months that are really busy, with just an occasional bit of time available for side projects. Feel free to contribute whenever you have the time and interest. 🙏🏽

**Comment by creste** [comment]
Thanks @barrywhart, I'll keep that in mind!  I appreciate your fast response and really enjoyed the new developer on-boarding experience for this project.  The documentation was accurate and concise, the code layout was very logical, and the tests were easy to modify and run.  You and the other maintainers have done well!
